<?xml version="1.0"?>

<launch>
  <arg name="namespace" default="wp_manipulator"/>
  <arg name="velocity_control_file" default="$(find aerial_manipulators_control)/config/wp_manipulator_velocity_control.yaml"/>
  <arg name="position_control_file" default="$(find aerial_manipulators_control)/config/wp_manipulator_position_control.yaml"/>

  <!-- Load joint controller configurations from YAML file to parameter server -->
  <rosparam file="$(arg velocity_control_file)" command="load"/>

  <node name="controller_spawner" pkg="controller_manager" type="spawner" respawn="false"
    output="screen" ns="$(arg namespace)"  args="joint_state_controller
    joint1_velocity_controller
    joint2_velocity_controller
    joint3_velocity_controller
    joint4_velocity_controller
    joint5_velocity_controller
        ">
    <remap from="/robot_description" to="$(arg namespace)/robot_description"/>
  </node>

  <!-- convert joint states to TF transforms for rviz, etc -->
  <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher"
    respawn="false" output="screen"  ns="$(arg namespace)">
    <remap from="/joint_states" to="$(arg namespace)/joint_states"/>
    <remap from="/robot_description" to="$(arg namespace)/robot_description"/>
  </node>

  <node name="JointPositionControl" pkg="aerial_manipulators_control" type="JointPositionControl" output="screen"  ns="$(arg namespace)" args="joint1_position_controller joint2_position_controller joint3_position_controller joint4_position_controller joint5_position_controller">
    <param name="param_file" value="$(arg position_control_file)"/>
  </node>

</launch>
